public/app/features/trails/AutomaticMetricQueries/query-generators/general/index.test.ts

```diff
@@ -0,0 +1,38 @@
+import * as queries from './queries';
+import * as rules from './rules';
+import { SUFFIX_TO_EXPECTED_PARAMS, forEachKeyValue } from './rules.test';
```

Comment:
I need to move these out of `rules.test` as it seems to cause tests to repeat.
Maybe something like
`__test__/cases.ts`
and
`__test__/util.ts`